graph/graph.go

```diff
@@ -242,18 +245,30 @@ func (graph *Graph) newTempFile() (*os.File, error) {
 	return ioutil.TempFile(tmp, "")
 }
 
-func bufferToFile(f *os.File, src io.Reader) (int64, error) {
-	n, err := io.Copy(f, src)
+func bufferToFile(f *os.File, src io.Reader, compress bool) (int64, digest.Digest, error) {
+	h := sha256.New()
+	w := io.MultiWriter(f, h)
+	if compress {
+		w = gzip.NewWriter(w)
+	}
+	_, err := io.Copy(w, src)
+	if wc, ok := w.(io.Closer); ok {
```

Comment:
If it's confusing; perhaps some docs describing the function to prevent future coders tripping over it?